go/protocols/materialize/transactor.go

```diff
@@ -300,6 +378,7 @@ func RunTransactions(
 
 		// `startCommit` may be nil to indicate a no-op commit.
 		var stateUpdate *pf.ConnectorState = nil
+		commitStartedAt = time.Now()
 		if startCommit != nil {
 			stateUpdate, ourCommitOp = startCommit(
 				stream.Context(), runtimeCheckpoint)
```

Comment:
I'm wondering if we can move the line below here so that it is never zero.
```
storedHistory[round%storedHistorySize] = storeIt.Total
```

Reply:
`storeIt.Total` will not be zero after `transactor.Store` returns - ref calls to `(*StoreIterator).Next` in `lifecycle.go`. The counts of 0 are from the array's initialization, where it is initialized at N values, but we haven't completed any transactions yet. We don't want it to look like a bunch 0 sized transactions have run, and a transaction of 0 documents is impossible, so they are skipped really just until N transactions have been run and the array buffer is "filled".

[`container/Ring`](https://pkg.go.dev/container/ring) would work for this too, but I don't think the ergonomics of it are great compared to just using a fixed size array.

Reply:
I updated some of the comments around this logic which may be helpful. LMK what you think.